examples/http.php

```diff
@@ -1,16 +1,74 @@
 <?php
 
+namespace http;
+
 use Castor\Attribute\AsTask;
+use Symfony\Component\Filesystem\Filesystem;
 
+use function Castor\fs;
+use function Castor\http_download;
+use function Castor\http_request;
 use function Castor\io;
-use function Castor\request;
 
 #[AsTask(description: 'Make HTTP request')]
-function httpRequest(): void
+function request(): void
 {
     $url = $_SERVER['ENDPOINT'] ?? 'https://example.com';
 
-    $response = request('GET', $url);
+    $response = http_request('GET', $url);
 
     io()->writeln($response->getContent());
 }
+
+#[AsTask(description: 'Download a file through HTTP')]
+function download(): void
+{
+    $downloadUrl = 'http://eu-central-1.linodeobjects.com/speedtest/100MB-speedtest';
+
+    if (isset($_SERVER['ENDPOINT'])) {
+        $downloadedFile = '2MB-dummy-file';
+        generateDummyFile($dummyFilePath = __DIR__ . '/../tests/Helper/fixtures/http/' . $downloadedFile);
+
+        $downloadUrl = $_SERVER['ENDPOINT'] . '/' . $downloadedFile;
+    }
+
+    $downloadedFilePath = '/tmp/castor-tests/examples/http-download-dummy-file';
+
+    try {
+        $response = http_download($downloadUrl, $downloadedFilePath, stream: false);
+
+        io()->writeln(
+            sprintf(
+                'Successfully downloaded file of size "%s" from url "%s" to "%s" with status code "%s"',
+                filesize($downloadedFilePath),
+                $downloadUrl,
+                $downloadedFilePath,
+                $response->getStatusCode()
+            )
+        );
+    } finally {
+        fs()->remove(array_filter([$downloadedFilePath, $dummyFilePath ?? null]));
+    }
+}
+
+function generateDummyFile(string $filename, int $sizeInMB = 2): void
```

Comment:
Simplified the code and relocated it to TaskTestCase, though I'm concerned about mixing test-related code with example code.

Reply:
Well my concern was legitimate :sweat_smile:  as examples run without access to test-related code: https://github.com/jolicode/castor/actions/runs/8951860409/job/24588633155?pr=433

Since it's now a 3 lines function I've inlined the code directly, makes the subject a non issue now IMO.